server/config/config.go

```diff
@@ -287,11 +287,18 @@ func DefaultConfig() *Config {
 }
 
 // GetConfig returns a fully parsed Config object.
-func GetConfig(v *viper.Viper) Config {
-	globalLabelsRaw := v.Get("telemetry.global-labels").([]interface{})
+func GetConfig(v *viper.Viper) (Config, error) {
+	globalLabelsRaw, ok := v.Get("telemetry.global-labels").([]interface{})
+	if !ok {
+		return *DefaultConfig(), fmt.Errorf("failed to parse global-labels config")
```

Comment:
In the case of an error, I would rather prefer returning an empty `Config{}`
```suggestion
		return Config{}, fmt.Errorf("failed to parse global-labels config")
```